src/internal/internalAPIs.ts

```diff
@@ -232,9 +232,13 @@ export function handleParentMessage(evt: DOMMessageEvent): void {
     const callback = GlobalVars.callbacks[message.id];
     if (callback) {
       callback.apply(null, message.args);
-
-      // Remove the callback to ensure that the callback is called only once and to free up memory.
-      delete GlobalVars.callbacks[message.id];
+      if (
+        !('isPartialResponse' in evt.data) ||
+        !(typeof evt.data.isPartialResponse === 'boolean') ||
+        !(evt.data.isPartialResponse === true)
+      ) {
+        delete GlobalVars.callbacks[message.id];
```

Comment:
Who would be responsible for cleaning up the callback in this case? Would the host be expected to send the final message with `isPartialResponse` set to false?

Reply:
That's right. Host is expected to send the final callback with either isPartialResponse set to false or without isPartialResponse param